Compiler warning C4819 in Japanese Windows

When I compile realm-core with MSBuild on Japanese Windows, MSBuild shows me warnings, like this.

```
E:\repo\realm-core\master\src\realm/alloc.hpp : warning C4819: ファイルは、現在のコード ページ (932) で表示できない文字を含んでいます。データの損失を防ぐために、ファイルを Unicode 形式で保存してください。 [E:\repo\realm-core\master\Visual Studio\Realm.vcxproj]
```

alloc.hpp has two non-ASCII characters.

1. https://github.com/realm/realm-core/blob/aa70c5c5771b5147a46adff22ac84660533d241c/src/realm/alloc.hpp#L315
2. https://github.com/realm/realm-core/blob/aa70c5c5771b5147a46adff22ac84660533d241c/src/realm/alloc.hpp#L317

alloc.hpp is encoded to UTF-8 without BOM. But Visual C++ treats a file which have no BOM as a file encoded to system encoding, for example Shift_JIS(932) in Japanese Windows.

I want you to fix this problem, because it is possible that they will be unable to compile realm-core in non-English Windows in the future.

Environment: Visual Studio 2015 (Japanese)

Version of Realm: aa70c5c5771b5147a46adff22ac84660533d241c

--- a/src/realm/alloc.hpp
+++ b/src/realm/alloc.hpp
@@ -312,9 +312,9 @@ inline ref_type to_ref(int_fast64_t v) noexcept
 
     // C++11 standard, paragraph 4.7.2 [conv.integral]:
     // If the destination type is unsigned, the resulting value is the least unsigned integer congruent to the source
-    // integer (modulo 2n where n is the number of bits used to represent the unsigned type). [ Note: In a two’s
+    // integer (modulo 2n where n is the number of bits used to represent the unsigned type). [ Note: In a two's
     // complement representation, this conversion is conceptual and there is no change in the bit pattern (if there is
-    // no truncation). — end note ]
+    // no truncation). - end note ]
     static_assert(std::is_unsigned<ref_type>::value,
                   "If ref_type changes, from_ref and to_ref should probably be updated");
     return ref_type(v);

